src/base_directories.rs

```diff
@@ -970,8 +1257,8 @@ mod test {
         let xd = BaseDirectories::with_env(
             "",
             "",
+            "test_files/user".to_string(),
             &*make_env(vec![
-                ("HOME", "test_files/user".to_string()),
```

Comment:
Because we get `home` from `std::env::home_dir()`, this actually doesn't do anything. Instead, in order to test this, we'd have to `std::env::set_var()`. But that would change the `HOME` env var for the main test process. In order to avoid that, I added a `test_home_environment` so that it can be tested in isolation with out affecting other tests that rely on `HOME`.